libdrgn/drgn.h.in

```diff
@@ -817,6 +817,22 @@ struct drgn_error *drgn_program_find_symbol_by_name(struct drgn_program *prog,
 						    const char *name,
 						    struct drgn_symbol **ret);
 
+/**
+ * Get all global and local symbols corresponding to the given name.
+ *
+ * @param[out] ret Pointer to a variable to be filled with an array of symbols.
+ * The resulting value is non-@c NULL on success, and has a length of @a @c
+ * *count pointers. Each symbol sohuld be freed with @ref drgn_symbol_destroy(),
+ * and the array itself should be freed with the standard free() function.
```

Comment:
Maybe we should hide the deallocation in a `void drgn_symbols_free(struct drgn_symbol **syms, size_t count)` function. Then users don't need to duplicate it and we also have the flexibility to change how the allocation works if we want.

Reply:
Adding onto this, I think it makes sense for `drgn_symbols_free()` to skip any `NULL` pointers within the `syms` array. This allows the caller to "take ownership" of any pointer by copying it and then setting its array entry to `NULL`, and I could directly use that in the Python binding rather than my static helper function.